cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java

```diff
@@ -61,13 +65,34 @@ public final void append(ILoggingEvent eventObject) {
       if (loggingContext == null) {
         return;
       }
+      addErrorCodeTags(eventObject);
       addExtraTags(eventObject, loggingContext);
     }
 
     LogMessage logMessage = new LogMessage(eventObject, loggingContext);
     appendEvent(logMessage);
   }
 
+  /**
+   * if event is for an exception deriving from ErrorCodeProvider, capture error group etc.
+   * @param event
+   */
+  private void addErrorCodeTags(ILoggingEvent event) {
+    if (event.getThrowableProxy() != null && event.getThrowableProxy() instanceof ThrowableProxy) {
+      Throwable throwable = ((ThrowableProxy) (event.getThrowableProxy())).getThrowable();
+      if (throwable != null && throwable instanceof ErrorTagProvider) {
```

Comment:
We should collect tags from all root causes that are providers

Reply:
We may also want to do some autodetection centrally, e.g. SocketException indicates it's network

Reply:
Added tags from all causes. For the SocketException, added a TODO.